### PR TITLE
Added extensions methods

### PR DESCRIPTION
**Pull request by GeertvanHorrik**

### CODE REVIEW

File: src/GitTools.Core.Tests/Extensions/StringExtensionsFacts.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        [TestCase("/develop", false)]
+        [TestCase("/master", false)]
+        [TestCase("/pr/25", true)]
+        [TestCase("/pull/25", true)]
+        [TestCase("/pull-requests/25", true)]
+        public void TheIsPullRequestMethod(string input, bool expectedValue)
+        {
+            var actualValue = input.IsPullRequest();
+            
+            Assert.AreEqual(expectedValue, actualValue);
```
Comment (JakeGinnivan) [inline src/GitTools.Core.Tests/Extensions/StringExtensionsFacts.cs]: I'm quite a fan of Shouldly for assertions. `input.IsPullRequest().ShouldBe(expectedValue)` will give quite good error messages.

File: src/GitTools.Core/Constants.cs [hunk 2]
```diff
@@ -1,4 +1,30 @@
 ﻿namespace GitTools
 {
-    // TODO: Constants go here
+    using System;
+    using LibGit2Sharp;
+
+    public static class TestValues
```
Comment (JakeGinnivan) [inline src/GitTools.Core/Constants.cs]: Wondering if we can make this class name better?

File: src/GitTools.Core/Constants.cs [hunk 3]
```diff
@@ -1,4 +1,30 @@
 ﻿namespace GitTools
```
Comment (JakeGinnivan) [inline src/GitTools.Core/Constants.cs]: `Testing` namespace?

File: src/GitTools.Core/Extensions/StringBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿namespace GitTools
+{
+    using System.Text;
+
+    public static class StringBuilderExtensions
+    {
+        //[StringFormatMethod("format")]
```
Comment (JakeGinnivan) [inline src/GitTools.Core/Extensions/StringBuilderExtensions.cs]: I think we should add the r# annotations nuget to make the usage of these light up?

File: src/GitTools.Core/Constants.cs (hunk 2 above)
Comment (GeertvanHorrik) [inline src/GitTools.Core/Constants.cs]: Feel free to come up with something better, as long as it's not Constants ;-)

File: src/GitTools.Core/Helpers/ProcessHelper.cs [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                    process = Process.Start(startInfo);
+                    process.PriorityClass = ProcessPriorityClass.Idle;
+                }
+            }
+
+            return process;
+        }
+
+        // http://csharptest.net/532/using-processstart-to-capture-console-output/
+        public static int Run(Action<string> output, Action<string> errorOutput, TextReader input, string exe, string args, string workingDirectory, params KeyValuePair<string, string>[] environmentalVariables)
```
Comment (JakeGinnivan) [inline src/GitTools.Core/Helpers/ProcessHelper.cs]: I think we should be using threads to read this. I should add an issue to fix this

File: src/GitTools.Core.Tests/Extensions/StringExtensionsFacts.cs (hunk 1 above)
Comment (GeertvanHorrik) [inline src/GitTools.Core.Tests/Extensions/StringExtensionsFacts.cs]: I'll give you this one. It's all a matter of preference, to be honest I think nunit provides really good feedback (I expected x, but got y). Not sure how shouldly does it but I'll do it anyway.

File: src/GitTools.Core/Testing/Extensions/GitTestExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+            ProcessHelper.Run(
+                o => output.AppendLine(o),
+                e => output.AppendLineFormat("ERROR: {0}", e),
+                null,
+                "git",
+                @"log --graph --abbrev-commit --decorate --date=relative --all",
+                repository.Info.Path);
+
+            Trace.Write(output.ToString());
```
Comment (JakeGinnivan) [inline src/GitTools.Core/Testing/Extensions/GitTestExtensions.cs]: We should return the string or log it to Info for this library?

File: src/GitTools.Core/Helpers/ProcessHelper.cs (hunk 5 above)
Comment (GeertvanHorrik) [inline src/GitTools.Core/Helpers/ProcessHelper.cs]: Probably. I just copy/pasted this.

File: src/GitTools.Core.Tests/Extensions/StringExtensionsFacts.cs (hunk 1 above)
Comment (JakeGinnivan) [inline src/GitTools.Core.Tests/Extensions/StringExtensionsFacts.cs]: Shouldly actually grabs the source code from your code to use in the assertion. It should give better error messages in all cases :)

File: src/GitTools.Core/Helpers/ProcessHelper.cs (hunk 5 above)
Comment (JakeGinnivan) [inline src/GitTools.Core/Helpers/ProcessHelper.cs]: Yep, not for this PR anyways. Just a note to myself really

**Comment by GeertvanHorrik** [comment]
@JakeGinnivan All changes I think should be made are made.

**Comment by JakeGinnivan** [comment]
Nice!

**Pull request merged**